File: vf_inventory_crm_be/app/schemas/customer.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class CustomerBase(BaseModel):
    name: str

    phone: Optional[str] = None

    email: Optional[str] = None

    address: Optional[str] = None

    is_active: bool = True


class CustomerCreate(CustomerBase):
    pass


class CustomerUpdate(BaseModel):
    name: Optional[str] = None

    phone: Optional[str] = None

    email: Optional[str] = None

    address: Optional[str] = None

    is_active: Optional[bool] = None


class CustomerResponse(CustomerBase):
    id: uuid.UUID

    created_at: datetime

    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True
    )

File: vf_inventory_crm_fe/lib/models/customer.dart
class Customer {
  final String id;
  final String name;
  final String? phone;
  final String? email;
  final String? address;
  final bool isActive;

  Customer({
    required this.id,
    required this.name,
    this.phone,
    this.email,
    this.address,
    required this.isActive,
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      id: json['id']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      phone: json['phone']?.toString(),
      email: json['email']?.toString(),
      address: json['address']?.toString(),
      isActive: json['is_active'] ?? true,
    );
  }
}
